polars/polars-core/src/chunked_array/ops/repeat_by.rs

```diff
@@ -50,6 +50,15 @@ impl RepeatBy for BooleanChunked {
 impl RepeatBy for Utf8Chunked {
     fn repeat_by(&self, by: &IdxCa) -> ListChunked {
         // TODO! dispatch via binary.
+        if (self.len() != by.len()) & (by.len() == 1) {
+            return self.repeat_by(&IdxCa::new(
+                "",
```

Comment:
This should be `self.name()`

Reply:
I added this. But another question. The bug is caused by the truncating behaviour this [`zip`](https://github.com/pola-rs/polars/pull/9023/files#diff-c3dc2c04f86517fbfe4d3c8c53aea5b99cafe3e322589daaef081da240a86ecfR64). So, the following case is not handled correctly:
```python
import polars as pl
from polars.testing import assert_frame_equal

df = pl.DataFrame(
    {
        "a": ["x", "y", "z"],
        "n": [3, 3, 3],
    }
)
expected = pl.DataFrame({"a": [["x", "x", "x"], ["y", "y", "y"], ["z", "z", "z"]]})

result = df.select(pl.col("a").repeat_by(by=pl.Series([2, 3])))
# This is result
# >>> df.select(pl.col("a").repeat_by(pl.Series([2, 3])))
# shape: (2, 1)
# ┌─────────────────┐
# │ a               │
# │ ---             │
# │ list[str]       │
# ╞═════════════════╡
# │ ["x", "x"]      │
# │ ["y", "y", "y"] │
# └─────────────────┘
assert_frame_equal(result, expected)  # Fails
```
I would argue that we panic if `(by.len() != 1) & (by.len() != self.len())`